go/vt/vtgate/evalengine/compiler_arithmetic.go

```diff
@@ -92,14 +100,15 @@ func (c *compiler) compileArithmeticAdd(left, right Expr) (ctype, error) {
 	if err != nil {
 		return ctype{}, err
 	}
+	skip1 := c.compileNullCheck1(lt)
```

Comment:
These are needed because arithmetic operations short circuit if the left hand side has a `nil` and it doesn't evaluable the right hand side then.